presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergLogicalPlanner.java

```diff
@@ -240,6 +247,836 @@ public void testFiltersWithPushdownDisable()
+                "(DATE '2015-01-13', 4), " +
+                "(DATE '2015-01-13', 5), " +
+                "(DATE '2015-05-15', 6), " +
+                "(DATE '2015-05-15', 7), " +
+                "(DATE '2020-02-21', 8), " +
+                "(DATE '2020-02-21', 9)";
+        assertUpdate(session, "INSERT INTO test_day_transform_date " + values, 12);
+        assertQuery(session, "SELECT * FROM test_day_transform_date", values);
+
+        String expected = "VALUES " +
+                "(NULL, 1, NULL, NULL), " +
+                "(DATE '1969-01-01', 1, DATE '1969-01-01', DATE '1969-01-01'), " +
+                "(DATE '1969-12-31', 1, DATE '1969-12-31', DATE '1969-12-31'), " +
+                "(DATE '1970-01-01', 1, DATE '1970-01-01', DATE '1970-01-01'), " +
+                "(DATE '1970-03-04', 1, DATE '1970-03-04', DATE '1970-03-04'), " +
+                "(DATE '2015-01-01', 1, DATE '2015-01-01', DATE '2015-01-01'), " +
+                "(DATE '2015-01-13', 2, DATE '2015-01-13', DATE '2015-01-13'), " +
+                "(DATE '2015-05-15', 2, DATE '2015-05-15', DATE '2015-05-15'), " +
+                "(DATE '2020-02-21', 2, DATE '2020-02-21', DATE '2020-02-21')";
+        assertQuery(session, "SELECT d_day, row_count, d.min, d.max FROM \"test_day_transform_date$partitions\"",
+                expected);
+
+        // Exercise non-pushdownable predicates
+        assertQuery(session, "SELECT * FROM test_day_transform_date WHERE day_of_week(d) = 3 AND b % 7 = 3",
+                "VALUES (DATE '1969-01-01', 10)");
+
+        assertPlan(session, "SELECT * FROM test_day_transform_date WHERE d IS NOT NULL",
+                output(exchange(strictTableScan("test_day_transform_date", identityMap("d", "b")))));
+
+        assertPlan(session, "SELECT * FROM test_day_transform_date WHERE d IS NULL",
+                output(exchange(strictTableScan("test_day_transform_date", identityMap("d", "b")))));
+
+        assertPlan(session, "SELECT * FROM test_day_transform_date WHERE d >= DATE '2015-01-13'",
+                output(exchange(strictTableScan("test_day_transform_date", identityMap("d", "b")))));
+
+        assertPlan(session, "SELECT * FROM test_day_transform_date WHERE CAST(d AS date) >= DATE '2015-01-13'",
+                output(exchange(strictTableScan("test_day_transform_date", identityMap("d", "b")))));
+
+        //TODO: d comparison with TIMESTAMP should be unwrapped
+        assertPlan(session, "SELECT * FROM test_day_transform_date WHERE d >= cast(TIMESTAMP '2015-01-13 00:00:00' as date)",
+                output(exchange(strictTableScan("test_day_transform_date", identityMap("d", "b")))));
+        assertPlan(session, "SELECT * FROM test_day_transform_date WHERE d > cast(TIMESTAMP '2015-01-13 00:00:00.001' as date)",
+                output(exchange(strictTableScan("test_day_transform_date", identityMap("d", "b")))));
+
+        // TODO: the following test cases could work after optimizer implementing unwrap the left function
```

Comment:
IMO I think we should file github issue for this, move these to a separate test function with an `@Ignore` and a comment linking to the github issue. Same for the rest of the left-unwrapped TODOs.

Reply:
Great idea, I will do it.